src/test/java/cz/muni/fi/pv286/mmt/representation/ArrayParserTest.java

```diff
@@ -0,0 +1,397 @@
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "{0x01, 0x02[}".getBytes(), "[0x01, 0x02]".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "[0x01, 0x02[]".getBytes(), "[0x01, 0x02]".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "(0x01, 0x02[)".getBytes(), "[0x01, 0x02]".getBytes());
+
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "{[, 0x01, 0x02}".getBytes(), "[0x01, 0x02]".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "[[, 0x01, 0x02]".getBytes(), "[0x01, 0x02]".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "([, 0x01, 0x02)".getBytes(), "[0x01, 0x02]".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "{0x01, 0x02, [}".getBytes(), "[0x01, 0x02]".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "[0x01, 0x02, []".getBytes(), "[0x01, 0x02]".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "(0x01, 0x02, [)".getBytes(), "[0x01, 0x02]".getBytes());
+
+        options.setBracketOption(BracketType.REGULAR_BRACKET);
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "{(0x01, 0x02}".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "[(0x01, 0x02]".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "((0x01, 0x02)".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "{0x01, 0x02(}".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "[0x01, 0x02(]".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "(0x01, 0x02()".getBytes(), "(0x01, 0x02)".getBytes());
+
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "{(, 0x01, 0x02}".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "[(, 0x01, 0x02]".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "((, 0x01, 0x02)".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "{0x01, 0x02, (}".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "[0x01, 0x02, (]".getBytes(), "(0x01, 0x02)".getBytes());
+        assertConversionThrows(options, ArrayParser.class, ArrayParser.class, "(0x01, 0x02, ()".getBytes(), "(0x01, 0x02)".getBytes());
```

Comment:
**suggestion:** Extract asserts into separate, ideally reusable method, to make this test more readable

This test is too long and very difficult to read. It would also be very difficult to find a mistake if this test was failing.